modules/api/src/main/java/com/intuit/wasabi/api/AssignmentsResource.java

```diff
@@ -152,18 +155,18 @@ public Response getAssignment(
         Assignment assignment = getAssignment(userID, applicationName, experimentLabel, context, createAssignment,
                 ignoreSamplingPercent, null, headers);
 
-        return httpHeader.headers().entity(toMap(assignment)).build();
+        return httpHeader.headers().entity(toSingleAssignmentResponseMap(assignment)).build();
     }
 
     private Assignment getAssignment(final User.ID userID, final Application.Name applicationName,
                                      final Experiment.Label experimentLabel, final Context context,
                                      final boolean createAssignment, final boolean ignoreSamplingPercent,
                                      final SegmentationProfile segmentationProfile, final HttpHeaders headers) {
-        Assignment assignment = assignments.getSingleAssignment(userID, applicationName, experimentLabel, context,
-                createAssignment, ignoreSamplingPercent, segmentationProfile, headers, null);
+        Assignment assignment = assignments.doSingleAssignment(userID, applicationName, experimentLabel, context,
+                createAssignment, ignoreSamplingPercent, segmentationProfile, headers);
 
         // This should not happen when createAssignment == true
```

Comment:
When createAssignment is TRUE then process will create a NEW ASSIGNMENT if it doesn't exists already.
When createAssignment is FALSE then process will NOT create a NEW ASSIGNMENT and only returns EXISTING_ASSIGNMENTs.